packages/flutter_tools/templates/ffi_plugin/README.md.tmpl

```diff
@@ -0,0 +1,85 @@
+# {{projectName}}
+
+{{description}}
+
+## Getting Started
+
+This project is a starting point for a Flutter 
+[FFI plugin](https://docs.flutter.dev/development/platform-integration/c-interop),
+a specialized package that includes native code directly invoked with Dart FFI.
+
+## Project stucture
+
+This template uses the following structure:
+
+* `src`: Contains the native source code, and a CmakeFile.txt file for building
+  that source code into a dynamic library
+
+* `lib`: Contains the Dart code that defines the API of the plugin, and which
+  calls into the native code using `dart:ffi`
+
+* platform folders (`android`, `ios`, `windows`, etc.): Contains the build files
+  for building and bundling the native code library with the platform application.
+
+## Buidling and bundling native code
+
+The `pubspec.yaml` specifies FFI plugins as follows.
+
+```yaml
+  plugin:
+    platforms:
+      some_platform:
+        ffiPlugin: true
+```
+
+This configuration invokes the native build for the various target platforms
+and bundles the binaries in Flutter applications using these FFI plugins.
+
+A plugin can have both FFI and method channels:
+
+```yaml
+  plugin:
+    platforms:
+      some_platform:
+        pluginClass: SomeName
+        ffiPlugin: true
+```
+
+The native build systems that are invoked by FFI (and method channel) plugins are:
+
+* For Android: Gradle, which invokes the Android NDK for native builds.
+  * See the documentation in android/build.gradle.
+* For iOS and MacOS: PodSpec.
```

Comment:
Podspec

But "Podspec" isn't a build system, it's a file that CocoaPods uses to generate Xcode; this would be sort of like saying "For Linux and Windows: CMakeLists.txt" below. Maybe say "Xcode, via CocoaPods"?